src/forms/Button.jsx

```diff
@@ -22,17 +22,17 @@ class Button extends React.PureComponent {
 		// since this functionality is built into button in order to enforce strict sizes on button icons
 		const props = { icon: icon && icon.props && icon.props.shape };
 
-		if (component !== 'button') {
+		if (component !== 'button' || 'a') {
```

Comment:
This logic isn't right. the second half of the statement is just 'a' which will always evaluate to true. I'm not sure what you actually want here.. maybe this `component !== 'button' && component !== 'a'`

Reply:
I think you should also add a test to make sure this warning is logged only when you want it to.